appscale/tools/agents/azure_agent.py

```diff
@@ -454,6 +532,19 @@ def create_or_update_vm_scale_sets(self, count, parameters, subnet):
     verbose = parameters[self.PARAM_VERBOSE]
     random_resource_name = Haikunator().haikunate()
 
+    # While autoscaling, look through existing scale sets to check if they have
+    # capacity to hold more vms. If they do, update the scale sets with additional
+    # vms. If not, then create a new scale set for them.
+    is_autoscale = parameters['autoscale_agent']
+    num_instances_to_add = count
+    if is_autoscale in ['True', True]:
+      instances_added = self.add_instances_to_existing_ss(
+        num_instances_to_add, parameters)
+      if num_instances_to_add > instances_added:
+        count = count - instances_added
```

Comment:
Yeah thats right, It was easier to let it modify the count because I wanted it to default to the regular logic of creating new scale sets once its exhausted the capacity of existing ones. But it makes sense to not to change the original count. I will fix that!